test/tuples_test.cpp

```diff
@@ -0,0 +1,130 @@
+#include "fixed_containers/tuples.hpp"
+
+#include <gtest/gtest.h>
+
+namespace fixed_containers::tuples::as_tuple_codegen
+{
+TEST(Tuples, AsTupleViewCodegenBranches)
+{
+    // `DISABLED_` would cause gtest to report that 1 test is disabled,
+    // so using this instead.
+    if constexpr ((false))
+    {
+        static constexpr std::size_t MAX_VARIABLE_COUNT = 400;
+
+        std::string variables{};
+        for (std::size_t i = 1; i <= MAX_VARIABLE_COUNT; i++)
+        {
+            if (!variables.empty())
+            {
+                variables += ", ";
+            }
+            variables += "m" + std::to_string(i);
+            std::cout << "    else if constexpr(FIELD_COUNT == " << i << ") {\n"
+                      << "        auto& [" << variables << "] = data;\n"
+                      << "        return std::tie(" << variables << ");\n"
+                      << "    }\n";
+        }
+    }
+}
+}  // namespace fixed_containers::tuples::as_tuple_codegen
+
+namespace fixed_containers
+{
+namespace
+{
+struct MyColors
```

Comment:
I don't think this is a meaningful name for this struct any more

Reply:
Renamed to `StructWithNestedStructs`